crates/rattler_package_streaming/tests/extract.rs

```diff
@@ -279,6 +280,29 @@ fn test_extract_flaky_conda(#[values(0, 1, 13, 50, 74, 150, 8096, 16384, 20000)]
     assert_matches::assert_matches!(result, ExtractError::IoError(_));
 }
 
+#[rstest]
+fn test_extract_data_descriptor_package_fails_streaming_and_uses_buffering() {
+    let package_path = "tests/resources/ca-certificates-2024.7.4-hbcca054_0.conda";
+
+    let temp_dir = Path::new(env!("CARGO_TARGET_TMPDIR"));
+    let target_dir = temp_dir.join("package_using_data_descriptors");
+    let result = extract_conda_via_streaming(File::open(package_path).unwrap(), &target_dir)
+        .expect_err("this should error out and not panic");
+
+    assert_matches::assert_matches!(
+        result,
+        ExtractError::ZipError(zip::result::ZipError::UnsupportedArchive(
+            "The file length is not available in the local header"
+        ))
+    );
+
+    let new_result =
+        extract_conda_via_buffering(File::open(package_path).unwrap(), &target_dir).unwrap();
+
+    insta::assert_snapshot!("new_result_sha256", &format!("{:x}", new_result.sha256));
+    insta::assert_snapshot!("new_result_md5", &format!("{:x}", new_result.md5));
```

Comment:
Please combine the two snapshot values in a single snapshot (otherwise you have to run the test twice to update the snapshots) and since the output is not large prefer an [inline snapshot](https://insta.rs/docs/snapshot-types/#inline-snapshots).

Reply:
Addressed in https://github.com/conda-incubator/rattler/pull/797/commits/16f584098cd141d193c3876e5e74f0ac0536fc77, please let me know if my approach makes sense!